src/app/course/_services/question-report.service.ts

```diff
@@ -0,0 +1,44 @@
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs';
+import {HttpClient} from '@angular/common/http';
+import {catchError} from 'rxjs/operators';
+import {ApiService} from "@app/_services/api.service";
+import {QuestionReport} from "@app/_models/question_report";
+
+@Injectable({
+    providedIn: 'root'
+})
+export class QuestionReportService {
+    constructor(
+        private http: HttpClient,
+        private apiService: ApiService) {
+    }
+
+    getReport(reportId: number): Observable<QuestionReport> {
+        const url = this.apiService.getURL('question-report', reportId);
+        return this.http
+            .get<QuestionReport>(url)
+            .pipe(catchError(this.apiService.handleError<QuestionReport>('Error occurred while fetching a report')));
+    }
+
+    sendReport(data: { user: number, question: number, report: string, report_details: string }): Observable<QuestionReport> {
```

Comment:
If you are going to use these anonymous types often, then just make it an actual type. Ideally, it should be the type you use in your form. If you don't always need to include user or question ID, then you can make them optional variables.

Reply:
We fixed the data type!